app_dart/lib/src/service/scheduler/files_changed_optimization.dart

```diff
@@ -66,25 +66,33 @@ final class FilesChangedOptimizer {
         log.warn('$refusePrefix: $reason');
         return FilesChangedOptimization.none;
       case SuccessfulFilesChanged(:final filesChanged):
-        var markdownOnly = true;
+        var noSourceImpact = true;
         for (final file in filesChanged) {
           if (file == 'DEPS' || file.startsWith('engine/')) {
             log.info(
               '$refusePrefix: Engine sources changed.\n${filesChanged.join('\n')}',
             );
             return FilesChangedOptimization.none;
           }
-          if (markdownOnly && p.posix.extension(file) != '.md') {
-            markdownOnly = false;
+          if (noSourceImpact &&
+              p.posix.extension(file) != '.md' &&
```

Comment:
at some point, not today, we'll probably want to refactor this to a chain filter or something.